vavr/src/main/java/io/vavr/PartialFunction.java

```diff
@@ -42,86 +43,116 @@
     long serialVersionUID = 1L;
 
     /**
-     * Unlifts a {@code totalFunction} that returns an {@code Option} result into a partial function.
-     * The total function should be side effect free because it might be invoked twice: when checking if the
-     * unlifted partial function is defined at a value and when applying the partial function to a value.
+     * Applies this function to the given argument and returns the result.
+     *
+     * @param t the argument
+     * @return the result of function application
      *
-     * @param totalFunction the function returning an {@code Option} result.
-     * @param <T> type of the function input, called <em>domain</em> of the function
-     * @param <R> type of the function output, called <em>codomain</em> of the function
-     * @return a partial function that is not necessarily defined for all input values of type T.
      */
-    static <T, R> PartialFunction<T, R> unlift(Function1<? super T, ? extends Option<? extends R>> totalFunction) {
-        return new PartialFunction<T, R>() {
-
-            private static final long serialVersionUID = 1L;
-
-            @Override
-            public R apply(T t) {
-                return totalFunction.apply(t).get();
```

Comment:
This was really eval. The main purpose of a PartialFunction is to provide _fast_ tests before applying the function.

Reply:
However, we don't really lose s.th. here. If we decided that we really need this functionality, we could add static methods to Option, e.g.

```java
final class Option<T> {
    static <T, R> PartialFunction<T, R> unlift(Function1<? super T, ? extends Option<? extends R>> totalFunction) {
        ...
    }
}
```

But for now I don't see the necessity.